docs/eeps/draft-embulk-system-properties-and-embulk-home.md

```diff
@@ -0,0 +1,144 @@
+* It does not need so powerful representation. A simple string-string key-value mapping would be sufficient.
+
+Embulk System Properties
+=========================
+
+The new system-wide configuration is designed as below, to satisfy the requirements above.
+
+* The new system-wide configuration is based on [`java.util.Properties`](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html), named "Embulk System Properties". `Properties` is not very rich, indeed. But, `Properties` can be loaded from a file only with the Java standard class library in the very beginning of a Java process.
+* A new concept of the "Embulk Home" directory is introduced. The Embulk Home directory takes over the older `~/.embulk/` directory that contained plugin installations. The Embulk Home directory includes the Embulk System Properties file, and continues to have plugin installations by default.
+* The Embulk Home directory and the Embulk System Properties are configured by a deterministic rule at runtime, based on the command-line options, the environment variables, the Embulk System Properties file, and the current working directory.
+
+The details follow.
+
+Setting the Embulk System Properties
+-------------------------------------
+
+The Embulk System Properties are set by the following methods, in priority from top to bottom.
+
+1. The command-line option `-X`, for example, `-Xproperty=value`.
+2. The Embulk System Properties file in the [Java `.properties` format](https://docs.oracle.com/javase/8/docs/api/java/util/Properties.html). The location of the Embulk System Properties file is identified based on the Embulk Home directory.
+
+Embulk Home
+------------
+
+The Embulk Home directory is determined at runtime, by the following rule, in priority from top to bottom.
+
+1. If an Embulk System Property `embulk_home` is set by the command-line option `-X`, it is the Embulk Home directory.
+    * It would be an absolute path, or a relative path from the current working directory.
+    * Ex. `java -jar embulk.jar -Xembulk_home=/var/tmp/foo run ...` (absolute path)
+    * Ex. `java -jar embulk.jar -Xembulk_home=.embulk2/bar run ...` (repative path)
+2. If an environment variable `EMBULK_HOME` is set, it is the Embulk Home directory.
+    * It must be an absolute path.
+    * Ex. `env EMBULK_HOME=/var/tmp/baz java -jar embulk.jar run ...` (absolute path)
+3. If neither 1 nor 2 matches, the Embulk Home directory is determined by searching from the current working directory as follows.
+    * The search starts from the current working directory, moving one by one toward the parent.
+    * In each directory, look for a directory, whose name is `.embulk/`, and which contains a regular file named `embulk.properties`.
+    * If such a directory `.embulk/` is found, it is the Embulk Home directory.
+    * If the current working directory is under the user's home directory (`user.home`), the search stops at the home directory.
```

Comment:
What do you think about adding Java property explicitly? It's little confuse Embulk property or not.

Reply:
Let me confirm your intention. Do you mean :

1. Adding some explanation in this EEP about the relationship with "Java (System) Property"?
2. Adding another implemented rule to consider Java System Property (not just Embulk System Properties) to determine Embulk System Properties and Embulk Home?
3. Something else?

Reply:
`user.home` is used this place only.
I'm not familiar with Java, When I first read this EEP, I didn't realize `user.home` is the Java property, not Embulk System property. 

For example, Changing to "`user.home`(Java Property)", is helpful for me.

Reply:
Ah, got it! That's a reasonable point. Let me fix it.

Reply:
Fixed it in dedeb3c8ce8a9cd35f6ff5a62bc45eb3eba2aa94